src/main/java/uk/org/okapibarcode/backend/Pdf417.java

```diff
@@ -1847,4 +1848,129 @@ public String toString() {
             return mode + "x" + length;
         }
     }
+
+    /**
+     * Splits up the data into a series of structured append PDF417 symbols which include
+     * segment macro data <code>total</code> and <code>position</code>.
+     * Input data will be assumed to be of the type set by {@link Pdf417#setDataType(Symbol.DataType)}.
+     *
+     * @param data     the data to encode
+     * @param template the PDF417 symbol template which will be used for all created
+     *                 symbols. The following properties will be ignored from the template:
+     *                 <ul>
+     *                      <li>{@link Pdf417#setContent(String)}</li>
+     *                      <li>{@link Pdf417#setContent(byte[])}</li>
+     *                      <li>{@link Pdf417#setStructuredAppendTotal(int)}</li>
+     *                      <li>{@link Pdf417#setStructuredAppendPosition(int)}</li>
+     *                 </ul>
+     * @throws OkapiException if no data or data is invalid
+     */
+    public static List<Pdf417> createStructuredAppendSymbols(String data, Pdf417 template) {
+        List<String> dataList = splitData(data, template);
+        return createStructuredAppendSymbols(dataList, template);
+    }
+
+    /**
+     * Overloaded method of {@link #createStructuredAppendSymbols(String, Pdf417)} which
+     * accepts binary data.
+     *
+     * <p><b>NOTE:</b>See linked method above for details.
+     */
+    public static List<Pdf417> createStructuredAppendSymbols(byte[] data, Pdf417 template) {
+        return createStructuredAppendSymbols(new String(data, ISO_8859_1), template);
+    }
+
+    private static List<String> splitData(String data, Pdf417 template) {
+        Pdf417 testSymbol = new Pdf417() {
+            @Override
+            protected void plotSymbol() {
+            } // expensive plotting is not required
+        };
+        clone(template, testSymbol);
+        testSymbol.setStructuredAppendTotal(2);
+
+        List<String> dataList = new ArrayList<>();
+        String remainingData = data;
+        while (!remainingData.isEmpty()) {
+            int low = 0;
+            int high = remainingData.length();
```

Comment:
@gredler Do you think we can limit the `high`? I.e. is there a way to calculate the maximum ever possible data amount for a specified symbol? That would allow us to optimize performance.

(Adding comment again there in order to make it visible after moving code around.)